core/src/main/java/org/infinispan/marshall/protostream/impl/SerializationContextRegistryImpl.java

```diff
@@ -43,6 +43,9 @@ public class SerializationContextRegistryImpl implements SerializationContextReg
 
    @Start
    public void start() {
+      global.addContextInitializer(new CommonTypesSchema());
+      global.addContextInitializer(new CommonContainerTypesSchema());
+      global.addContextInitializer(new UserContextInitializerImpl());
```

Comment:
@ryanemerson Can you take a look to confirm this is alright?

Do we even need a global context if we are adding all these things?

Reply:
This is correct :+1:

> Do we even need a global context if we are adding all these things?

I'm not really sure what you mean here. We need to add these types to the global schema as this is the context used by the ProtoStreamTranscoder. Unfortunately we have two SerializationContexts for remote query, because `ProtobufMetadataManagerImpl` needs to add additional configuration options to the context used for queries, so it's not possible to just use the global context everywhere.